src/_guides/language/sound-problems.md

```diff
@@ -466,29 +486,48 @@ filterValues((x) => (x [!as String!]).contains('Hello'));
 
 <hr>
 
-<a id="common-errors-and-warnings"></a>
-## Runtime errors
+### Incorrect type inference
 
-Dart enforces a sound type system. Roughly, this means
-you can't get into a situation where the value stored in a variable is
-different from the variable's static type. Like most modern statically
-typed languages, Dart accomplishes this with a combination of static
-(compile-time) and dynamic (runtime) checking.
+```nocode
+error - ...
+```
 
-For example, the following type error is detected at compile-time:
+On rare occasions, Dart's enhaced type inference may infer the wrong type for function literal arguments in a generic constructor invocation.
+This primarily affects `Iterable.fold`
+
+#### Example
+
+In this code, type inference will infer that `a` has a type of `Null`:
 
 {:.fails-sa}
+<?code-excerpt "lib/... ()" replace=""?>
 {% prettify dart tag=pre+code %}
-List<int> numbers = [1, 2, 3];
-List<String> [!string = numbers!];
+void main() {
```

Comment:
Perhaps we don't need to include the `main()` function scaffolding in these example, so it's easier to focus on the relevant code.